File: flow-tests/test-npm-only-features/test-default-theme/frontend/src/template-with-client-side-imports.js
import {PolymerElement} from '@polymer/polymer/polymer-element.js';
import {html} from '@polymer/polymer/lib/utils/html-tag.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';

class TemplateWithClientSideImports extends PolymerElement {
  static get template() {
    return html`
       <vaadin-button>Button</vaadin-button>
    `;
  }
  static get is() {
    return 'template-with-client-side-imports'
  }
    
}
  
customElements.define(TemplateWithClientSideImports.is, TemplateWithClientSideImports);

File: flow-tests/test-npm-only-features/test-default-theme/src/main/java/com/vaadin/flow/testonlynpmfeatures/defaulttheme/TemplateWithClientSideImports.java
/**
 * Copyright (C) 2024 Vaadin Ltd
 *
 * This program is available under Vaadin Commercial License and Service Terms.
 *
 * See <https://vaadin.com/commercial-license-and-service-terms> for the full
 * license.
 */
package com.vaadin.flow.testonlynpmfeatures.defaulttheme;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.dependency.NpmPackage;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.templatemodel.TemplateModel;

@JsModule("./src/template-with-client-side-imports.js")
@NpmPackage(value = "@vaadin/vaadin-button", version = "2.2.0")
@Tag("template-with-client-side-imports")
public class TemplateWithClientSideImports
        extends PolymerTemplate<TemplateModel> {

}
